data-runtime/src/main/java/io/micronaut/data/runtime/mapper/BeanIntrospectionMapper.java

```diff
@@ -93,33 +90,13 @@ R map(@NonNull D object, @NonNull Class<R> type) throws InstantiationException {
                     if (property.getType().isInstance(v)) {
                         property.set(instance, v);
                     } else if (Iterable.class.isAssignableFrom(property.getType())) {
-                        Object value = property.get(instance);
-                        Collection<?> collection;
                         if (v instanceof Collection) {
-                            collection = (Collection<?>) v;
+                            property.set(instance, v);
                         } else if (v instanceof Iterable) {
-                            collection = new ArrayList<>(CollectionUtils.iterableToList((Iterable) v));
-                        } else if (v.getClass().isArray()) {
-                            Object[] arr = (Object[]) v;
-                            collection = Arrays.asList(arr);
-                        } else if (v instanceof java.sql.Array) {
-                            Object[] arr;
-                            try {
-                                arr = (Object[]) ((java.sql.Array) v).getArray();
-                            } catch (SQLException e) {
-                                throw new DataAccessException("Unable to read SQL array", e);
-                            }
-                            collection = Arrays.asList(arr);
-                        } else {
-                            collection = Collections.singleton(v);
-                        }
-                        if (value instanceof Collection) {
-                            ((Collection) value).addAll(collection);
```

Comment:
This was the issue, we were adding property values twice to the value already initialized in the constructor.